crates/sui-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -3549,14 +3549,17 @@ impl AuthorityPerEpochStore {
         authority_metrics
             .consensus_handler_cancelled_transactions
             .inc_by(cancelled_txns.len() as u64);
+
+        let max_cost_us = shared_object_congestion_tracker.max_cost();
+        let max_cost_randomness_us = shared_object_using_randomness_congestion_tracker.max_cost();
         authority_metrics
             .consensus_handler_max_object_costs
             .with_label_values(&["regular_commit"])
-            .set(shared_object_congestion_tracker.max_cost() as i64);
+            .observe(max_cost_us as f64 / 1_000.0);
```

Comment:
I'd leave this unmodifed and fix the buckets, because the "/1000" scalar makes no sense for modes that are not ExecutionTimeEstimate. Then we can adjust units as needed in grafana

(For that matter, the "_us" in the variable names above is incorrect because the commit handler is agnostic as to congestion control units.)

Reply:
Sure, I don't mind, but aren't we going to delete the old modes anyway? And we don't really care about this metric for the older ones.

Reply:
I can see that argument. If you really want to make it specific to this mode I think it should be f64 seconds to match the convention for the other time-based histogram metrics (latency etc) - but nbd